pkg/client/unversioned/conditions.go

```diff
@@ -54,11 +54,15 @@ func JobHasDesiredParallelism(c ExtensionsInterface, job *extensions.Job) wait.C
 		}
 
 		// desired parallelism can be either the exact number, in which case return immediately
-		if job.Status.Active == *job.Spec.Parallelism {
+		if job.Spec.Parallelism != nil && job.Status.Active == *job.Spec.Parallelism {
 			return true, nil
 		}
 		// otherwise count successful
-		progress := *job.Spec.Completions - job.Status.Active - job.Status.Succeeded
+		completions := 1
+		if job.Spec.Completions != nil {
```

Comment:
I can live with null-checks, but I don't want to apply defaults in two places. That seems wrong to me.